File: src/types/agent-context.ts
export interface AgentContextWorktree {
  repoId: string;
  repoName: string;
  path: string;
  branch?: string | null;
  head?: string | null;
}

export interface WorkspaceAgentContext {
  workspaceId: string;
  primaryPath: string;
  linkedWorktrees: AgentContextWorktree[];
  promptPreamble: string;
}

export interface RepoMapEntry {
  path: string;
  kind: string;
  symbols: string[];
}

export interface RepoMapMeta {
  version: number;
  branch: string;
  refName: string;
  commitHash: string;
  generatedAt: string;
}

export interface RepoMap {
  version: number;
  generatedAt: string;
  branch: string;
  refName: string;
  commitHash: string;
  entries: RepoMapEntry[];
}

export interface WorkspaceContextItem {
  label: string;
  path?: string | null;
  kind: string;
  priority: number;
  chars: number;
  included: boolean;
  trimmed: boolean;
}

export interface WorkspaceContextPreview {
  workspaceId: string;
  repoRoot: string;
  status: string;
  defaultBranch: string;
  refName: string;
  commitHash: string;
  generatedAt?: string | null;
  approxChars: number;
  maxChars: number;
  trimmed: boolean;
  items: WorkspaceContextItem[];
  promptContext: string;
  warning?: string | null;
}

File: src-tauri/src/models/agent_context.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct AgentContextWorktree {
    pub repo_id: String,
    pub repo_name: String,
    pub path: String,
    pub branch: Option<String>,
    pub head: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceAgentContext {
    pub workspace_id: String,
    pub primary_path: String,
    pub linked_worktrees: Vec<AgentContextWorktree>,
    pub prompt_preamble: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceContextPreview {
    pub workspace_id: String,
    pub repo_root: String,
    pub status: String,
    pub default_branch: String,
    pub ref_name: String,
    pub commit_hash: String,
    pub generated_at: Option<String>,
    pub approx_chars: usize,
    /// `0` when Forge does not apply a character budget to repo context.
    pub max_chars: usize,
    pub trimmed: bool,
    pub items: Vec<WorkspaceContextItem>,
    pub prompt_context: String,
    pub warning: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceContextItem {
    pub label: String,
    pub path: Option<String>,
    pub kind: String,
    pub priority: u8,
    pub chars: usize,
    pub included: bool,
    pub trimmed: bool,
}
